src/main.rs

```diff
@@ -344,16 +349,39 @@ pub enum RecordsOut {
 }
 
 impl RecordsIn {
-    pub async fn into_reader(&self) -> std::io::Result<Box<dyn AsyncBufRead + Send + Unpin>> {
+    pub async fn into_lines_stream(
+        &self,
+    ) -> std::io::Result<Pin<Box<dyn Stream<Item = std::io::Result<String>> + Send>>> {
         match self {
-            RecordsIn::File(path) => Ok(Box::new(BufReader::new(File::open(path).await?))),
-            RecordsIn::Stdin => Ok(Box::new(BufReader::new(tokio::io::stdin()))),
+            RecordsIn::File(path) => {
+                let file = File::open(path).await?;
+                Ok(Box::pin(LinesStream::new(
+                    tokio::io::BufReader::new(file).lines(),
+                )))
+            }
+            RecordsIn::Stdin => Ok(Box::pin(stdio_lines_stream(std::io::stdin()))),
```

Comment:
lets use std for stdout as well then?

Reply:
We can. I don't see any issues there.

Reply:
Taking this back. `Stdout` is a writer. Completely different.

Reply:
okay yeah makes sense